### ISSUE TITLE
High CPU load when there are tons of fsx files

### ISSUE THREAD

**Issue by johlrich** [georgewfraser/fsharp-language-server#8]
When using paket's ability to generate load scripts, I'll have thousands of fsx files under the .paket/load folder that all get scanned when the extension loads causing vscode and the language server to eat lots of resources + fans to turn on.

![image](https://user-images.githubusercontent.com/1655793/41330590-6aa1bafc-6ea2-11e8-8564-532dfa6cf303.png)

Is it possible to set folders to be ignored?

**Comment by georgewfraser** [comment]
Hmm wow. Clearly, we should not analyze .fsx files until they are open...in the meantime, you can make a file `.vscode/settings.json` that looks like:

```json
{
    "files.exclude": {
        "evil-folder/": true
    }
}

**Comment by johlrich** [comment]
Thanks for the quick reply. I just tried that but no luck after adding that and restarting vscode. 

The folders did stop appearing in the file view, but the extension was still scanning them. Since it's all generated stuff, it wasn't a big deal to just removed them for now.

**Comment by georgewfraser** [comment]
The right solution is definitely to load .fsx files lazily

**Comment by georgewfraser** [comment]
Fixed by 85907c98a90bd336caccadc0a3357fbda72aa6ed
